raft/logger.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package raft
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+type Logger interface {
+	// EnableTimestamps enables the logger to print timestamp
+	// prefix. It is useful to disable it when using journald.
+	EnableTimestamps()
```

Comment:
I don't think this makes sense as a member of the Logger interface. In glog, for instance, timestamps are always used and cannot be turned off. Instead, I would expose the `logger` type as `StandardLogger`, and applications that want timestamps in the logs can do something like 
```
    l := log.New(os.Stderr, "", 0)
    l.SetFlags(l.Flags() | log.Ldate | log.Ltime)
    raft.SetLogger(raft.NewStandardLogger(l))
```

Actually, do we need the log.Logger object at all? I understand why users would want to switch from `log` to `loggo` or `glog`, but it's not clear that it's useful to switch between different `log.Logger` objects. If we just used the default log functions instead of a separate log.Logger, then applications could just use `log.SetFlags()` once; they wouldn't need to turn timestamps on separately for their main app and for raft.